docs/core/knowledge-bases.rst

```diff
@@ -0,0 +1,581 @@
+``InMemoryKnowledgeBase``:
+
+.. code-block:: python
+
+    knowledge_base = InMemoryKnowledgeBase("data.json")
+
+Every object in your knowledge base should have "name" and "id" field.
+If that is not the case, please read the section :ref:`customize_in_memory_knowledge_base`.
+
+
+Defining the NLU Data
+~~~~~~~~~~~~~~~~~~~~~
+
+In this section
+
+- we are going to introduce a new intent, ``query_knowledge_base``.
+- we are going to annotate ``mention`` entities so that our model detects indirect mentions of objects like "the
+  first one".
+- we will use synonyms (:ref:`entity_synonyms`) extensively.
+
+To be able to understand that the user wants to retrieve some information from the knowledge base, you need to define
+a new intent, for example, called ``query_knowledge_base``.
+The intent should contain all kind of user requests.
+
+Let's look at an example:
+
+.. code-block:: md
+
+    ## intent:query_knowledge_base
+    - what [restaurants](object_type:restaurant) can you recommend?
+    - list some [restaurants](object_type:restaurant)
+    - can you name some [restaurants](object_type:restaurant) please?
+    - can you show me some [restaurant](object_type:restaurant) options
+    - list [German](cuisine) [restaurants](object_type:restaurant)
+    - do you have any [mexican](cuisine) [restaurants](object_type:restaurant)?
+    - do you know the [price range](attribute:price-range) of [that one](mention)?
+    - what [cuisine](attribute) is [it](mention)?
+    - do you know what [cuisine](attribute) the [last one](mention:LAST) has?
+    - does the [first one](mention:1) have [outside seating](attribute:outside-seating)?
+    - what is the [price range](attribute:price-range) of [Berlin Burrito Company](restaurant)?
+    - what is with [I due forni](restaurant)?
+    - can you tell me the [price range](attribute) of [that restaurant](mention)?
+    - what [cuisine](attribute) do [they](mention) have?
+     ...
+
+The above example just shows examples related to the restaurant domain.
+You should add examples for every object type that exists in your knowledge base.
+
+All user requests can be divided into two categories:
+(1) The user wants to obtain a list of objects of a specific type or (2) the user wants to know about a certain
+attribute of an object.
+The ``ActionQueryKnowledgeBase`` can handle both of those requests.
+Other requests, such as comparison between objects, are currently not supported.
+
+Another thing you may have noticed is, that we marked different kind of entities in the NLU data.
+If you want to use ``ActionQueryKnowledgeBase``, you need to specify the following entities:
+
+- ``object_type``: Whenever the user is talking about a specific object type from your knowledge base, the type should
+  be marked as entity in our NLU data. Use :ref:`entity_synonyms` to map, for example, "restaurants" to the correct
+  object type listed in the knowledge base, e.g. "restaurant".
```

Comment:
Do they also need to add the 
```
##synonym: 1
- the first one

##synonym: outside-seating
- outside seating
```
etc. to the NLU data?

Reply:
You can, but you don't need to. The bot is working just fine without defining it. As explained in the `entity_synonyms` section, two ways exists to express synonyms. I guess we can leave it as it is.

Reply:
Got it. Just didn't know exactly how synonyms work 😃